src/AzSK.ADO/Framework/Abstracts/MetaInfoProvider.ps1

```diff
@@ -0,0 +1,284 @@
+			}
+		}
+
+		if ($ResourceTypeName -in ([ResourceTypeName]::VariableGroup, [ResourceTypeName]::All))
+		{
+			if (!$this.varGroupSTDetails) {
+				$this.varGroupSTDetails = [ConfigurationManager]::LoadServerConfigFile("VariableGroupSTData.json");
+			}
+        }
+        if ($ResourceTypeName -eq "ServiceTree")
+		{
+			if (!$this.serviceTreeDetails) {
+				$this.serviceTreeDetails = [ConfigurationManager]::LoadServerConfigFile("ServiceTreeData.json");
+			}
+		}
+    }
+    
+    [PSObject] FetchServiceAssociatedResources($svcId, $projectName, $ResourceTypeName)
+    {
+        $rsrcList = $null;
+        if ($this.bUseADOInfoAPI -eq $true -or $this.CheckIsADOInfoAPIEnabled())
+        {
+            #TODO: Look at cleaning up these multiple "-in" checks across the API_call-v-Policy_Repo cases...
+            #TODO-PERF: For now we are erring on the side of avoiding multiple network calls...revisit based on observed pattern of -svcid <xyz> usage
+            $qs = "?svcId={0}" -f $svcId
+            $rsrcList = $this.CallADOInfoAPI($qs);
+        }
+        else 
+        {
+            $this.FetchMappingFiles($ResourceTypeName);
+
+            $buildList = @{};
+            $releaseList = @{};
+            $svcConnList = @{};
+            $varGroupList = @{};
+            $agentPoolList = @{};
+
+            if ($this.buildSTDetails) {
+                $buildList = $this.buildSTDetails.Data | Where-Object { ($_.serviceId -eq $svcId) -and ($_.projectName -eq $projectName) }
+            }
+            if ($this.releaseSTDetails) {
+                $releaseList = $this.releaseSTDetails.Data | Where-Object { ($_.serviceId -eq $svcId) -and ($_.projectName -eq $projectName) }
+            }
+            if ($this.svcConnSTDetails) {
+                $svcConnList = $this.svcConnSTDetails.Data | Where-Object { ($_.serviceId -eq $svcId) -and ($_.projectName -eq $projectName) }
+            }
+            if ($this.agtPoolSTDetails) {
+                $agentPoolList = $this.agtPoolSTDetails.Data | Where-Object { ($_.serviceId -eq $svcId) -and ($_.projectName -eq $projectName) }
+            }
+            if ($this.varGroupSTDetails) {
+                $varGroupList = $this.varGroupSTDetails.Data | Where-Object { ($_.serviceId -eq $svcId) -and ($_.projectName -eq $projectName) }
+            } 
+            
+            $rsrcList = @{
+                Builds = $buildList
+                Releases = $releaseList
+                SvcConns = $svcConnList
+                VarGroups = $varGroupList
+                AgentPools = $agentPoolList
+            }
+
+        }
+        return $rsrcList; 
+    }
+    
+    [PSObject] FetchResourceMappingWithServiceData($rscId, $projectName, $resourceTypeName)
+    {
+        $serviceTree = $null;
+        try 
+        {
+            #check if adoinfoapi is enabled in org-policy file 
+            if ($this.bUseADOInfoAPI -eq $true -or $this.CheckIsADOInfoAPIEnabled())
```

Comment:
if not enabled, the isADOInfoAPIEnabled method will be called everytime a bug is logged.

Reply:
removed call isADOInfoAPIEnabled